File: ptra-server/src/utils/models/campaign.model.ts
import { CampaignType } from '../enum/campaign-type.enum';
import { DiscountType } from '../enum/discount-type.enum';

export default class CampaignModel {
  id: number;
  title: string;
  type: CampaignType;
  description: string;
  discountType: DiscountType;
  categoryId: number;

  constructor(
    id: number,
    title: string,
    type: CampaignType,
    description: string,
    categoryId: number,
    discountType: DiscountType,
  ) {
    this.id = id;
    this.title = title;
    this.description = description;
    this.type = type;
    this.categoryId = categoryId;
    this.discountType = discountType;
  }
}

File: ptra-mobile/lib/features/home/pages/cart/data/models/campaign_model.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import '../../domain/domain.dart';

part 'campaign_model.freezed.dart';
part 'campaign_model.g.dart';

@freezed
class CampaignModel with _$CampaignModel implements ICampaignEntity {
  factory CampaignModel({
    required final int id,
    required final String title,
    required final String description,
    required final int type,
    required final int categoryId,
    required final int discountType,
  }) = _CampaignModel;

  factory CampaignModel.fromJson(Map<String, dynamic> json) => _$CampaignModelFromJson(json);
}
